src/panels/lovelace/cards/hui-map-card.js

```diff
@@ -0,0 +1,258 @@
+          <div id="map"></div>
+        </div>
+      </ha-card>
+
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_drawEntities'
+      },
+      _config: Object,
+      isPanel: {
+        type: Boolean,
+        reflectToAttribute: true
+      }
+    };
+  }
+
+  constructor() {
+    super();
+    this._debouncedResizeListener = debounce(this._resetMap.bind(this), 100);
+  }
+
+  ready() {
+    super.ready();
+
+    if (!this._config || this.isPanel) {
+      return;
+    }
+
+    this.$.root.style.paddingTop = this._config.aspect_ratio || '100%';
+  }
+
+  setConfig(config) {
+    if (!config) {
+      throw new Error('Error in card configuration.');
+    }
+
+    this._configEntities = processConfigEntities(config.entities);
+    this._config = config;
+  }
+
+  getCardSize() {
+    let ar = this._config.aspect_ratio || '100%';
+    ar = ar.substr(0, ar.length - 1);
+    return 1 + Math.floor(ar / 25) || 3;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // Observe changes to map size and invalidate to prevent broken rendering
+    // Uses ResizeObserver in Chrome, otherwise window resize event
+    if (typeof ResizeObserver === 'function') {
+      this._resizeObserver = new ResizeObserver(() => this._debouncedResizeListener());
+      this._resizeObserver.observe(this.$.map);
+    } else {
+      window.addEventListener('resize', this._debouncedResizeListener);
+    }
+
+    if (!this._map) {
+      this._map = setupLeafletMap(this.$.map);
+    }
+
+    this._drawEntities(this.hass);
+
+    setTimeout(() => {
+      this._resetMap();
+      this._fitMap();
+    }, 1);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    if (this._resizeObserver) {
+      this._resizeObserver.unobserve(this.$.map);
+    } else {
+      window.removeEventListener('resize', this._debouncedResizeListener);
+    }
+  }
+
+  _resetMap() {
+    if (!this._map) {
+      return;
+    }
+    this._map.invalidateSize();
+  }
+
+  _fitMap() {
+    if (this._mapItems.length === 0) {
+      this._map.setView(
+        new Leaflet.LatLng(this.hass.config.core.latitude, this.hass.config.core.longitude),
+        14
+      );
+    } else {
+      const bounds = new Leaflet.latLngBounds(this._mapItems.map(item => item.getLatLng()));
+      this._map.fitBounds(bounds.pad(0.5));
+    }
+  }
+
+  _drawEntities(hass) {
+    const map = this._map;
+    if (!map) {
+      return;
+    }
+
+    if (this._mapItems) {
+      this._mapItems.forEach(marker => marker.remove());
+    }
+    const mapItems = this._mapItems = [];
+
+    // for (let entity of this._configEntities) {
+    this._configEntities.forEach((entity) => {
+      const entityId = entity.entity;
+      if (!(entityId in hass.states)) {
+        return;
+      }
+      const stateObj = hass.states[entityId];
+      const title = computeStateName(stateObj);
+      const attr = stateObj.attributes;
```

Comment:
```js
const { latitude, longitude } = stateObj.attributes;
```

Reply:
We also need `radius`, `passive`, `entity_picture`, `icon`, etc.

Reply:
Then add those. The code is similar to

```js
const latitude = stateObj.attributes.latitude
// etc
```